### PR TITLE
Show validation errors when creating/updating nodes

### PR DESCRIPTION
**Pull request by NikolausC**

### CODE REVIEW

File: app/controllers/nodes_controller.rb [hunk 1]
```diff
@@ -55,12 +55,19 @@ def update
...
       @node_form = NodeForm.new(node_form_params)
+
       if @node_form.save
         format.html { redirect_to node_path(@node.parent_id), flash: { success: "Node updated" }}
         format.json { render json: @node, status: :ok }
       else
-        format.html { redirect_to edit_node_path(@node.id), flash: { danger: "Failed to update node" }}
-        format.json { render json: @node.errors, status: :unprocessable_entity }
+        format.html {
+          flash[:error] = []
```
Comment (jbeast) [inline app/controllers/nodes_controller.rb]: Is this necessary? It's a little odd having flash[:error] being an array. Perhaps convert the errors to a sentence e.g. `errors.full_messages.to_sentence` instead

File: app/controllers/nodes_controller.rb [hunk 2]
```diff
@@ -55,12 +55,19 @@ def update
...
+          redirect_to edit_node_path(@node.id)
```
Comment (jbeast) [inline app/controllers/nodes_controller.rb]: Redirect should come last in an action

File: app/forms/node_form.rb [hunk 3]
```diff
@@ -49,8 +53,13 @@ def self.node_permitted(node, permission_type, groups)
 
   def create_objects
     ActiveRecord::Base.transaction do
-      node = Node.create!(name: name, cost_code: cost_code, description: description, parent_id: parent_id, owner: @owner)
-      node.permissions.create!(convert_permissions(@owner))
+      node = Node.new(name: name, cost_code: cost_code, description: description, parent_id: parent_id, owner: @owner)
+      if node.save
+        node.permissions.create!(convert_permissions(@owner))
+      else
+        @node = node
```
Comment (jbeast) [inline app/forms/node_form.rb]: This is very side affecty. You could just set the initial `node` as an instance variable i.e. `@node = Node.new(...)`

File: app/forms/node_form.rb [hunk 4]
```diff
@@ -59,10 +68,14 @@ def create_objects
...
-      node.update_attributes!(name: name, cost_code: cost_code, description: description, parent_id: parent_id)
-      node.permissions.destroy_all
-      node.set_permissions
-      node.permissions.create!(convert_permissions(node.owner))
+      if node.update_attributes(name: name, cost_code: cost_code, description: description, parent_id: parent_id)
+        node.permissions.destroy_all
+        node.set_permissions
+        node.permissions.create!(convert_permissions(node.owner))
+      else
+        @node = node
```
Comment (jbeast) [inline app/forms/node_form.rb]: Again, this isn't necessary. Also, `@node` won't be set if anything other than `update_attributes` fails.

I'd put it back to the way it was but like the other example, the `node` could instead be an instance variable e.g. `@node = Node.find(id)`

**Comment by NikolausC** [comment]
Think this still needs an alteration to create_objects, please can @jbeast check?

Cheers

**Pull request merged**